user/lookup_unix.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build darwin dragonfly freebsd !android,linux netbsd openbsd solaris
+// +build cgo
+
+package user
```

Comment:
os/user doesn't provide anything for looking up groups, and I couldn't find anything in the standard library that does group lookup, or even any actively maintained project either.

This whole section is there for username groupname lookup in the 'ls -l' style output returned by ReadDir. It's easy enough to remove this package, but when connecting to this sftp server via the 'sftp' program, output of ls -l will look strange, showing uid/gid instead of ownername groupname. Everything else should still work.

I don't believe group lookup should be part of the sftp package.

For the moment I'll remove this package entirely. Perhaps a group package, which has only the 'group' functions, and using os/user for user lookup is the best way to proceed.